sklearn/utils/estimator_checks.py

```diff
@@ -67,7 +68,9 @@
                 'OrthogonalMatchingPursuit', 'PLSCanonical', 'PLSRegression',
                 'RANSACRegressor', 'RadiusNeighborsRegressor',
                 'RandomForestRegressor', 'Ridge', 'RidgeCV']
-
+MULTI_OUTPUT_CLASSIFIER = ['DecisionTreeClassifier', 'ExtraTreeClassifier',
+                           'ExtraTreesClassifier', 'KNeighborsClassifier',
+                           'OneVsRestClassifier', 'RandomForestClassifier']
```

Comment:
I think we can avoid listing these. Calling fit on an estimator should either accept multi-output data and behave correctly, or should raise a ValueError with message "bad input shape ..." (try `check_X_y([[0]], [[0, 1]])`). So we can run the check on all estimators.

Reply:
@jnothman I've rebased this PR #14865 to finish it. On this comment I'm not sure to understand how you would handle this then.
Are you recommending:
```
try:
     y_pred = classifier.fit(X_train, y_train).predict(X_test)
     y_pred_list_of_lists = classifier.fit(
        X_train, y_train_list_of_lists).predict(X_test)
except ValueError:
     # classifier doesn't support multi-output
      return

# followed by checks
```
?

Reply:
My suggestion was that, if there is an error, you check that it as an appropriate message.

Also, when @herilalaina was working on this, we had not introduced estimator tags which may be another way to control this: assert the behaviour specified by the tags of that estimator